`Basic ESQL usage output (telemetry) non-snapshot version` is also failing. I don't understand why release test `esql.features` needs to be updated for a snapshot-only feature? //cc @luigidellaquila 

```
REPRODUCE WITH: ./gradlew ":x-pack:plugin:yamlRestTest" --tests "org.elasticsearch.xpack.test.rest.XPackRestIT.test {p0=esql/60_usage/Basic ESQL usage output (telemetry) non-snapshot version}" -Dtests.seed=3327970C418BA24F -Dbuild.snapshot=false -Dtests.jvm.argline="-Dbuild.snapshot=false" -Dlicense.key=x-pack/license-tools/src/test/resources/public.key -Dtests.locale=frr-Latn-DE -Dtests.timezone=Asia/Ashgabat -Druntime.java=24
XPackRestIT > test {p0=esql/60_usage/Basic ESQL usage output (telemetry) non-snapshot version} FAILED
    java.lang.AssertionError: Failure at [esql/60_usage:142]: field [esql.features] doesn't have length [25]
    Expected: <25>
         but: was <26>
        at __randomizedtesting.SeedInfo.seed([3327970C418BA24F:BB73A8D6EF77CFB7]:0)
        at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.executeSection(ESClientYamlSuiteTestCase.java:522)
        at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.test(ESClientYamlSuiteTestCase.java:482)
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
...
```

--- a/x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/esql/60_usage.yml
+++ b/x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/esql/60_usage.yml
@@ -139,7 +139,7 @@ setup:
   - do: {xpack.usage: {}}
   - match: { esql.available: true }
   - match: { esql.enabled: true }
-  - length: { esql.features: 25 }
+  - length: { esql.features: 26 }
   - set: {esql.features.dissect: dissect_counter}
   - set: {esql.features.drop: drop_counter}
   - set: {esql.features.eval: eval_counter}
@@ -165,6 +165,7 @@ setup:
   - set: {esql.features.fork: fork_counter}
   - set: {esql.features.rrf: rrf_counter}
   - set: {esql.features.completion: completion_counter}
+  - set: {esql.features.sample: sample_counter}
   - length: { esql.queries: 3 }
   - set: {esql.queries.rest.total: rest_total_counter}
   - set: {esql.queries.rest.failed: rest_failed_counter}
@@ -220,7 +221,7 @@ setup:
   - gt: {esql.functions.to_long: $functions_to_long}
   - match: {esql.functions.coalesce: $functions_coalesce}
   - gt: {esql.functions.categorize: $functions_categorize}
-  - length: {esql.functions: 131} # check the "sister" test above for a likely update to the same esql.functions length check
+  - length: {esql.functions: 132} # check the "sister" test above for a likely update to the same esql.functions length check
 
 ---
 took:

